src/libs/antares/study/parts/thermal/container.cpp

```diff
@@ -99,10 +96,9 @@ uint PartThermal::prepareClustersInMustRunMode()
     do
     {
         mustContinue = false;
-        auto end = list.end();
-        for (auto i = list.begin(); i != end; ++i)
+        for (auto i = list.begin(); i != list.end(); ++i)
```

Comment:
We have to keep the range loop with an iterator i because of the : 
```C++
std::shared_ptr<ThermalCluster> cluster = list.detach(i);
```
We should try (some day) to get rid of this iterator. For later.